src/Builder/ArrayExpressionBuilder.php

```diff
@@ -109,24 +108,28 @@ protected function buildPlaceholders(ExpressionInterface $expression, array &$pa
 
             if ($item instanceof ExpressionInterface) {
                 $placeholders[] = $this->queryBuilder->buildExpression($item, $params);
-                continue;
+            } else {
+                $placeholders[] = $this->queryBuilder->bindParam($item, $params);
             }
-
-            $placeholders[] = $this->queryBuilder->bindParam($item, $params);
         }
 
         return $placeholders;
     }
 
-    private function unnestArrayExpression(ArrayExpression $expression, mixed $value): ArrayExpression
+    /**
+     * @psalm-param ArrayExpression $expression
+     */
+    private function unnestArrayExpression(ExpressionInterface $expression, mixed $value): ArrayExpression
```

Comment:
If we are waiting here `ArrayExpression`, then better use type hint for it. As before.

And in other places it is similar.

Reply:
Can the `$expression` be  instance of something like `ArrayExpressionInterface` but not instance of `ArrayExpression`?

Reply:
Changed ExpressionInterface to ArrayExpression.
I think no extra check needed, it will throw an exception in one of private methods.